bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/TypeChecker.java

```diff
@@ -3213,11 +3213,17 @@ private static boolean hasFillerValue(Type type, List<Type> unanalyzedTypes) {
         if (type == null) {
             return true;
         }
-        if (type.getTag() < TypeTags.RECORD_TYPE_TAG &&
-                !(type.getTag() == TypeTags.CHAR_STRING_TAG || type.getTag() == TypeTags.NEVER_TAG)) {
+
+        int typeTag = type.getTag();
+        if (TypeTags.isXMLTypeTag(typeTag)) {
+            return (typeTag == TypeTags.XML_TAG || typeTag == TypeTags.XML_TEXT_TAG);
```

Comment:
Do we need parentheses here ex.  return typeTag == TypeTags.XML_TAG || typeTag == TypeTags.XML_TEXT_TAG;